### PR TITLE
Don't warn on PodInitializing errors for deck /logs

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @Katharine @BenTheElder

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Oops, the logger already had the id.

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -746,7 +746,16 @@ func handleLog(lc logClient) http.HandlerFunc {
 		log, err := lc.GetJobLog(job, id)
 		if err != nil {
 			http.Error(w, fmt.Sprintf("Log not found: %v", err), http.StatusNotFound)
-			logger.WithError(err).Warning("Log not found.")
+			logger := logger.WithError(err)
+			msg := "Log not found."
+			if strings.Contains(err.Error(), "PodInitializing") {
```
Comment (BenTheElder) [inline prow/cmd/deck/main.go]: seems slightly brittle /shrug
/lgtm
/meow space

**Pull request merged**